README.md

```diff
@@ -1,2 +1,152 @@
+* **The Shared Memory Wrappers**: abbreviated `shm_wrapper` and `shm_wrapper_aux`, these two are tools that facilitate the efficient communication between the above three processes. Think of these wrappers as the "glue" that holds the three processes together, and lets them talk to each other.
+* **The Logger**: this is the tool that Runtime uses to gather all the logs generated at various places in the code (including by student code) and outputs them to a terminal window, to a file, to Dawn over the network, or to some combination of the three.
+* **The Device Code**: codenamed "`lowcar`" <sup id="return2">[2](#footnote2)</sup> this is the set of all the code on the Arduinos that directly control an individual device that is connected to the robot. All of this code is collectively called the "lowcar library".
+
+This README will not go into each of these parts into exhaustive detail; explanations for each part can be found in the corresponding folder in the repo for the interested reader. However, we will describe briefly the data that flows between the various parts, and the manner in which that data is sent:
+
+* `net_handler` communicates with both Dawn and Shepherd. It receives start and stop commands, input data for the coding challenge, and information about the gamepad state. It sends log messages, data about the connected devices, and output data for the coding challenge. This data is sent over both TCP and UDP connections (depending on the type of data) to Shepherd and Dawn, and the data is packaged using Google Protocol Buffers.
+* `dev_handler` and `net_handler` are connected via `shm_wrapper`. This connection is used for the `net_handler` to fetch the most recent state of the connected devices, which is then sent to Dawn for the students to see.
+* `executor` and `net_handler` are connected via `shm_wrapper_aux` and a FIFO pipe. The connections are used to pass information about the coding challenge (both inputs to the functions and the student's outputs), and for `executor` to know what to run at any given time (autonomous mode, teleop mode, coding challenges, or idle).
+* `executor` and `dev_handler` are connected via `shm_wrapper`. This connection is used for `executor` to send commands to the attached devices (tell a motor to run at a certain speed, move a servo to certain position, etc.), and for `dev_handler` to serve `executor` with the device data that it needs to run the student code.
+* `dev_handler` communicates with `lowcar` via serial connection. This connection is used to poll for new devices, detect when devices have disconnected, and send device data and commands between the Raspberry Pi and the Arduinos.
+* All three processes output log messages through the `logger` tool. If logs are to be sent over the network to Dawn, those logs are put into a FIFO pipe which is opened by `net_handler`, where the logs are processed and sent to Dawn.
+
+## Contents
```

Comment:
At least for me, I think it would make more sense if you combined the "Overall Structure" and "Contents" section. Main reasons are they have a large overlap in what they say. And additionally, most details for the contents can/should be found in their respective READMEs.

Reply:
Yeah, reading it again I think the "Contents" sections is mostly redundant. I'll delete it.

Reply:
Should combine it, cus you left out systemd and runtime_util I think.

Reply:
👌